ui/app/src/components/DashletCard/DashletCard.tsx

```diff
@@ -69,61 +91,98 @@ export function DashletCard(props: DashletCardProps) {
       (!renderCardHeader ? cardHeaderHeight : 0)
     : UNDEFINED;
   return (
-    <Card sx={{ borderLeft: 5, borderLeftColor, ...sxs?.card }}>
+    <Card
+      sx={{
+        display: 'flex',
+        flexDirection: 'column',
+        borderLeft: 5,
+        borderLeftColor,
+        ...sxs?.card,
+        ...(isMaximized && {
+          position: 'absolute',
+          zIndex: 5,
+          top: 0,
+          bottom: 0,
+          left: 0,
+          right: 0
+        })
+      }}
+    >
       {/* region Header */}
       {renderCardHeader && (
         <CardHeader
           title={title}
           titleTypographyProps={{ variant: 'h6', component: 'h2' }}
-          action={headerAction}
+          action={
+            <>
+              {maximizable && (
+                <IconButton onClick={() => updateMaximized()}>
+                  {isMaximized ? <CloseFullscreenOutlinedIcon /> : <OpenInFullOutlinedIcon />}
+                </IconButton>
+              )}
+              {headerAction}
+              {!isMaximized && (
+                <IconButton onClick={() => setIsExpanded(!isExpanded)}>
+                  {isExpanded ? <ExpandLessOutlinedIcon /> : <ExpandMoreOutlinedIcon />}
+                </IconButton>
+              )}
```

Comment:
I'm thinking... let's make the whole header collapse/expand. Just make sure the actual button clicks stop the event so they don't unintentionally collapse the widget.

What if we wrap the CardHeader in CardActionArea.

Let's get rid of the icon/button all together; though not as discoverable, I'm hoping the action area highlight would help with that.